hazelcast/src/main/java/com/hazelcast/sql/impl/expression/string/LikeFunction.java

```diff
@@ -0,0 +1,204 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.string;
+
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.expression.TriExpression;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.hazelcast.sql.impl.expression.string.StringExpressionUtils.asVarchar;
+
+/**
+ * LIKE string function.
+ */
+public class LikeFunction extends TriExpression<Boolean> implements IdentifiedDataSerializable {
+
+    private static final long serialVersionUID = 4157617157954663651L;
+
+    /** Single-symbol wildcard in SQL. */
+    private static final char ONE_SQL = '_';
+
+    /** Multi-symbol wildcard in SQL. */
+    private static final char MANY_SQL = '%';
+
+    /** Single-symbol wildcard in Java. */
+    private static final String ONE_JAVA = ".";
+
+    /** Multi-symbol wildcard in Java. */
+    private static final String MANY_JAVA = "(?s:.*)";
+
+    /** Special characters which require escaping in Java. */
+    private static final String ESCAPE_CHARACTERS_JAVA = "[]()|^-+*?{}$\\.";
+
+    private transient State state;
+
+    public LikeFunction() {
+        // No-op.
+    }
+
+    private LikeFunction(Expression<?> source, Expression<?> pattern, Expression<?> escape) {
+        super(source, pattern, escape);
+    }
+
+    public static LikeFunction create(Expression<?> source, Expression<?> pattern, Expression<?> escape) {
+        return new LikeFunction(source, pattern, escape);
+    }
+
+    @SuppressFBWarnings(value = "NP_BOOLEAN_RETURN_NULL", justification = "SQL has three-valued boolean logic")
+    @Override
+    public Boolean eval(Row row, ExpressionEvalContext context) {
+        String source = asVarchar(operand1, row, context);
+
+        if (source == null) {
+            return null;
+        }
+
+        String pattern = asVarchar(operand2, row, context);
+
+        if (pattern == null) {
+            return null;
+        }
+
+        String escape = operand3 != null ? asVarchar(operand3, row, context) : null;
```

Comment:
if escape was provided and its value is null, it should return null

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/17379/commits/bf96a792b021e758f55a09c628d72d04b8d98594